### DISCUSSION TITLE
Document API for tenant onboarding

### DISCUSSION THREAD

**Discussion by AffiTheCreator** [awslabs/aws-saas-boost#78] [Q&A] [answered]
Is there any documentation on how to use the API to provision a tenant fro example?

**Comment by brtrvn** [comment, marked as answer]
Please add a feature request for this and we'll get it prioritized.

The short version to get you going:
1. Get the API Gateway stage URL for the public API from the API Gateway console. Go to API Gateway -> Select the public API -> Click on Stages from the left-hand menu -> Click on v1 in the list of stages -> Copy the Invoke URL
2. Execute an HTTP `POST` to INVOKE_URL/users/token with a body containing a valid username and password (your admin user and password)
`{"username": "admin", "password": "my-password"}`
3. Capture the value of the `accessToken` from the response to use as your JWT bearer token in subsequent requests
4. Execute a HTTP `POST` to INVOKE_URL/tenants with a body containing your onboarding request object and set the `Authentication` header value to `Bearer: value-of-access-token-from-step-3`
`{"name": "API Onboarded Tenant"}`
Optionally include `subdomain`
Optionally include `planId` for a billing plan subscription
Optionally override compute setting defaults by including `computeSize` (S, M, L, XL), `minCount`, `maxCount`

**Reply by PoeppingT** [reply]
https://github.com/awslabs/aws-saas-boost/issues/81
